### PR TITLE
A guided tour created with drive.js

### PR DESCRIPTION
**Pull request by altheaFeu**
Crafted using drive.js, this guided tour enable users to generate an engaging, interactive, and step-by-step tutorial experience

### CODE REVIEW

**Comment by Gustry** [comment]
Looks great !

Do you have a GIF or a PNG you can include in your PR and in your README.md ?

Side question, did you consider https://shepherdjs.dev/ ? We were thinking about this one.

**Comment by altheaFeu** [comment]
> Side question, did you consider https://shepherdjs.dev/ ? We were thinking about this one.

I wasn't aware of Shepherd before. I updated the script to:
- Replace driver.js with shepherd.js
- Add a gif in the readme.md.

However, this script doesn't handle actions and multiple highlights. It's a very simple and basic tour guide.

File: library/ui/driver_tutorial/README.md [hunk 1]
```diff
@@ -0,0 +1,17 @@
+# Interactive Step-by-Step Guided Tours
+![guided-tour](https://github.com/altheaFeu/lizmap-javascript-scripts/assets/123597411/4e2f95eb-28f5-4f11-b1c6-5db2901d1e05)
```
Comment (Gustry) [inline library/ui/driver_tutorial/README.md:2]:
```suggestion
![guided-tour](./demo.gif)
```

Can you host in the git repository ?

File: library/ui/driver_tutorial/README.md [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+## Introduction
+
+This script empowers you to craft a simple interactive guided tour using `shepherd.js`.
```
Comment (Gustry) [inline library/ui/driver_tutorial/README.md:5]:
```suggestion
This script empowers you to craft a simple interactive guided tour using [shepherd.js](https://shepherdjs.dev/).
```

File: library/ui/driver_tutorial/README.md [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+
+The initial popup design for the presentation draws inspiration from the metadata information popup available at:
+[Popup Metadata Info](https://github.com/3liz/lizmap-javascript-scripts/blob/master/library/ui/popup_metadata_info).
```
Comment (Gustry) [inline library/ui/driver_tutorial/README.md:8]:
```suggestion
[Popup Metadata Info](../library/ui/popup_metadata_info).
```

File: library/ui/driver_tutorial/tour.js [hunk 4]
```diff
@@ -0,0 +1,186 @@
...
+            tour.on('cancel', dismissTour);
+
+            // Start the tour
+            if (!localStorage.getItem('shepherd-tour')) {
+                tour.start();
+            }
+        };
+
+        script.onerror = function () {
+            console.error('Erreur lors du chargement de shepherd.js');
```
Comment (Gustry) [inline library/ui/driver_tutorial/tour.js:181]: Translate in english

**Review by Gustry — APPROVED** [review]
Minor comments.
Thanks for the contribution.

> I wasn't aware of Shepherd before. I updated the script to:

Sorry, I didn't expect you to rewrite your example using sheperd, I was just curious ;-) We did some experiments on this library for now.

**Comment by altheaFeu** [comment]
@Gustry I added a new UI that enables users to change the opacity of all layers that are checked in the layer tab.

**Comment by altheaFeu** [comment]
@Gustry, I have implemented a script that fetches data from a URL and displays it within a right-docked interface. This script allows users to view data that is not presented in the layer dock or attribute table but not external to the project (data_table.js).

**Comment by Gustry** [comment]
Thanks for your contribution @altheaFeu !
Can you look at comments above, on this page ?
Like translating all sentences to English, store GIF on the repository ?
Can you add the new directory in the main README.md as well ?

**Comment by altheaFeu** [comment]
@Gustry I made the changes as asked

**Comment by altheaFeu** [comment]
Hi @Gustry,
As I'll be proposing several tools over the next few days, I'm closing this pull request and creating a new one from my branch named "tour-guide" to avoid any confusion.

**Comment by altheaFeu** [comment]
Hi @Gustry,
As I'll be proposing several tools over the next few days, I'm closing this pull request and creating a new one from my branch named "tour-guide" to avoid any confusion.

**Pull request closed without merging**